src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -374,6 +374,12 @@ private async Task OnBuildUpdateAsync(IProjectVersionedValue<BuildUpdate> update
         Assumes.True(_state is WorkspaceState.Initialized);
         Assumes.NotNull(_buildProgressRegistration);
 
+        // The Roslyn workspace context is created when the first evaluation data arrives.
+        // It's possible that build data arrives before evaluation data, in which case
+        // the Roslyn context would be null. To prevent problems, we wait for evaluation
+        // data to have been processed at least once before continuing.
+        await _hasEvaluationData.Task;
```

Comment:
It looks like this might fault the block and report a error? Previously, we'd not fire up the language service for projects that had bad language data, (NoTargets for example would fall into this I think).

Reply:
I just ran some tests with `NoTargets` and didn't see anything problematic from the user's perspective, but I agree this should be investigated. I would have expected it to fail and either fault the project or silently report an NFE.

Filed https://github.com/dotnet/project-system/issues/8343 to track this.

Reply:
Set one of the properties its reading during init to empty.